File: packages/@simpli/cli-scaffold/generator/template/src/model/response/PrincipalResp.ts
/**
 * PrincipalResp
 * @author martinlabs CRUD generator
 */
import Principal from '@/model/Principal'
import {Resource} from '@/app/http/Resource'
import Tag from '@/model/Tag'
import GrupoDoPrincipal from '@/model/GrupoDoPrincipal'
import {ResponseSerialize} from '@/helpers/http.helper'
import {ID, TAG} from '@/types/app'

export default class PrincipalResp extends Resource {
  readonly $endpoint: string = '/Crud/Principal{/id}'

  @ResponseSerialize(Principal)
  principal: Principal = new Principal()

  @ResponseSerialize(GrupoDoPrincipal)
  allGrupoDoPrincipal: GrupoDoPrincipal[] = []

  @ResponseSerialize(Tag)
  allTag: Tag[] = []

  get $id() {
    return this.principal.$id
  }

  set $id(val: ID) {
    this.principal.$id = val
  }

  get $tag() {
    return this.principal.$tag
  }

  set $tag(val: TAG) {
    this.principal.$tag = val
  }
}

File: packages/@simpli/cli-server/generator/template/src/main/java/br/com/martinlabs/usecase/crud/response/PrincipalResp.kt
package br.com.martinlabs.usecase.crud.response

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty
import br.com.martinlabs.usecase.model.Principal
import br.com.martinlabs.usecase.model.GrupoDoPrincipal
import br.com.martinlabs.usecase.model.Tag

@ApiModel(value = "PrincipalResp")
class PrincipalResp {

    var principal: Principal? = null

    @ApiModelProperty(value = "Possible GrupoDoPrincipal values")
    var allGrupoDoPrincipal: MutableList<GrupoDoPrincipal>? = null

    @ApiModelProperty(value = "Possible Tag values")
    var allTag: MutableList<Tag>? = null
}
